controller_manager:
  ros__parameters:
    update_rate: 1000  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    franka_robot_state_broadcaster:
      type: franka_robot_state_broadcaster/FrankaRobotStateBroadcaster
    
    franka_robot_model_broadcaster:
      type: franka_robot_state_broadcaster/FrankaRobotModelBroadcaster

    multi_mode_controller:
      type: multi_mode_controller/MultiModeController

franka_robot_state_broadcaster:
  ros__parameters:
    arm_id: panda

franka_robot_model_broadcaster:
  ros__parameters:
    arm_id: panda
    
multi_mode_controller:
  ros__parameters:
    arm_count: 1
    arm_1:
      arm_id: panda
      translational_stiffness: [3000, 3000, 10000]
      rotational_stiffness:    [300, 300, 300]
      damping_ratio: 0.7
    controllers: ["panda_joint_impedance_controller", "panda_cartesian_impedance_controller"]
    resources:
        panda_joint_impedance_controller: ["panda"]
        panda_cartesian_impedance_controller: ["panda"]
    start_controllers:
        # names: ["panda_joint_impedance_controller"]
        # resources:
        #     panda_joint_impedance_controller: ["panda"]
        names: ["panda_cartesian_impedance_controller"]
        resources:
            panda_cartesian_impedance_controller: ["panda"]